micropip/manager.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any, List
+
+from micropip import package_index
+
+
+class Manager:
```

Comment:
Do we like the name `Manager`? It's a bit vague but seems alright to me. We should consider whether we can come up with something better. @jaraco maybe you have some thoughts?

Reply:
I don't love "Manager", but it's adequate.

I notice below the `repodata_` prefix on a couple of member variables, suggesting that maybe this object is representing `RepoData` or `State` or `RepositoryState` or `Repository` or `Repo` (and maybe the `repodata_` prefix can be removed). I also see that "repodata" is a legacy artifact and "lock" is relevant, so maybe `LockState` or some variant.

Since I see actions about `.install`, and `.freeze`, I'm thinking this is relevant to an `Environment` or `Site`.

I'm liking `Repository` or `Environment` best, but I'd be okay with any option. Feel free to pick/promote one you like best.

Am I right in thinking this class is an internal implementation detail (not part of a public API)? If so, the naming isn't critical and can be changed later.

Reply:
My thoughts were that eventually this class will become representative of the current public API, with additional extensibility such as dependency injection as discussed in #112, although I would guess having a separate exposed layer and keeping the `Manager` internal is also a feasible choice. (Assuming that the most common case of a user wanting to instantiate their own `Manager` without accessing the current public APIs would be to utilize such extensibility)

I was thinking that `Client` or `Micropip` could suit it better but `Manager` seemed closest to what it was going to do: state management. I do agree its a vague name, and `Repository` also sounds alright too.

Reply:
Yep, I plan to make this class public. People would do `mgr = micropip.Manager(); mgr.install(...)`.

I prefer `Manager` or `Environment`. The former emphasizes that this is a package manager of sorts, while the latter recognizes that each instance is like a virtual environment of sorts.

Let's go with `Manager` for now. We still have time before implementing all these APIs and making this class public.

Reply:
Maybe call it `PackageManager` then? or `???Environment`? I feel like we could use an extra word in the name.

Reply:
`PackageManager` sounds good to me.